booster/library/Booster/JsonRpc.hs

```diff
@@ -455,11 +466,108 @@ respond stateVar =
+                    Log.logDebug $ "Error internalising cterm" <> Text.pack (show patternError)
+                    pure $
+                        Left $
+                            RpcError.backendError $
+                                RpcError.CouldNotVerifyPattern
+                                    [ patternErrorToRpcError patternError
+                                    ]
+                (Right (patL, substitutionL, unsupportedL), Right (patR, substitutionR, unsupportedR)) -> do
+                    unless (null unsupportedL && null unsupportedR) $ do
+                        Log.logWarnNS
+                            "booster"
+                            "Implies: aborting due to unsupported predicate parts"
+                        unless (null unsupportedL) $
+                            Log.logOtherNS
+                                "booster"
+                                (Log.LevelOther "ErrorDetails")
+                                (Text.unlines $ map prettyPattern unsupportedL)
+                        unless (null unsupportedR) $
+                            Log.logOtherNS
+                                "booster"
+                                (Log.LevelOther "ErrorDetails")
+                                (Text.unlines $ map prettyPattern unsupportedR)
+                    let doTracing =
+                            Flag $
+                                any
+                                    (fromMaybe False)
+                                    [ req.logSuccessfulSimplifications
+                                    , req.logFailedSimplifications
+                                    ]
+                        -- apply the given substitution before doing anything else
+                        substPatL =
+                            Pattern
+                                { term = substituteInTerm substitutionL patL.term
+                                , constraints = Set.map (substituteInPredicate substitutionL) patL.constraints
+                                , ceilConditions = patL.ceilConditions
+                                }
+                        substPatR =
+                            Pattern
+                                { term = substituteInTerm substitutionR patR.term
+                                , constraints = Set.map (substituteInPredicate substitutionR) patR.constraints
+                                , ceilConditions = patR.ceilConditions
+                                }
+
+                    case matchTerms Booster.Pattern.Match.Implies def substPatR.term substPatL.term of
+                        MatchFailed (SubsortingError sortError) ->
+                            pure . Left . RpcError.backendError . RpcError.ImplicationCheckError . RpcError.ErrorOnly . pack $
+                                show sortError
+                        MatchFailed{} ->
+                            doesNotImply (sortOfPattern substPatL) req.antecedent.term req.consequent.term
```

Comment:
i guess this would matter if e.g. the antecedent goes to #bottom? i guess here i am assuming things are defined and the constraints don't contradict each other